packages/react/src/components/GroupedList/GroupedListV2.base.tsx

```diff
@@ -0,0 +1,604 @@
+    (item: IHeaderGroupedItem, flattenedIndex: number): React.ReactNode => {
+      const group = item.group;
+
+      const headerProps = {
+        ...groupProps!.headerProps,
+        ...getDividerProps(item.group, flattenedIndex),
+        key: group.key,
+        groupedListId: item.groupId,
+        ariaLevel: group.level ? group.level + 1 : 1,
+        ariaSetSize: groups ? groups.length : undefined,
+        ariaPosInSet: flattenedIndex !== undefined ? flattenedIndex + 1 : undefined,
+      };
+
+      return (
+        <GroupItem
+          render={onRenderHeader}
+          defaultRender={renderGroupHeader}
+          item={item}
+          selection={selection}
+          eventGroup={events.current}
+          props={headerProps}
+        />
+      );
+    },
+    [onRenderHeader, groupProps, groups, getDividerProps, selection],
+  );
+
+  const renderShowAll = React.useCallback(
+    (item: IShowAllGroupedItem, flattenedIndex: number): React.ReactNode => {
+      const group = item.group;
+      const groupShowAllProps = {
+        ...groupProps!.showAllProps,
+        ...getDividerProps(group, flattenedIndex),
+        key: group.key ? `${group.key}-show-all` : undefined,
+      };
+
+      return onRenderShowAll(groupShowAllProps, renderGroupShowAll);
+    },
+
+    [onRenderShowAll, groupProps, getDividerProps],
+  );
+
+  const renderFooter = React.useCallback(
+    (item: IFooterGroupedItem, flattenedIndex: number): React.ReactNode => {
+      const group = item.group;
+      const groupFooterProps = {
+        ...groupProps!.footerProps,
+        ...getDividerProps(group, flattenedIndex),
+        key: group.key ? `${group.key}-footer` : undefined,
+      };
+
+      return onRenderFooter(groupFooterProps, renderGroupFooter);
+    },
+    [onRenderFooter, groupProps, getDividerProps],
+  );
+
+  const renderItem = React.useCallback(
+    (item: IGroupedItem, flattenedIndex: number): React.ReactNode => {
+      if (item.type === 'header') {
+        return renderHeader(item, flattenedIndex);
+      } else if (item.type === 'showAll') {
+        return renderShowAll(item, flattenedIndex);
+      } else if (item.type === 'footer') {
+        return renderFooter(item, flattenedIndex);
+      } else if (item.type === 'item') {
+        const level = item.group.level ? item.group.level + 1 : 1;
+        return onRenderCell(level, item.item, item.itemIndex ?? flattenedIndex);
+      }
+    },
+    [onRenderCell, renderHeader, renderShowAll, renderFooter],
+  );
+
+  return (
+    <FocusZone
+      direction={FocusZoneDirection.vertical}
+      data-automationid="GroupedList"
+      data-is-scrollable="false"
+      role="presentation"
+      {...focusZoneProps}
+      shouldEnterInnerZone={shouldEnterInnerZone}
+      className={css(classNames.root, focusZoneProps.className)}
+    >
+      <List
+        ref={listRef}
+        role={role}
+        items={listView}
+        onRenderCellConditional={renderItem}
```

Comment:
In the end, you don't seem to be rendering conditioned based on scroll state. Can you just use `onRenderItem` after all?

Reply:
It's being used to conditionally render group footers. Because whether a footer is rendered or not is determined by the group footer render function having the conditional logic in the render loop seems the right choice to me. I experimented with having it in the `flattenItems` function which works but felt like an odd fit.

Reply:
I would personally like to avoid modifying `List` just to make this work, though. I think that allocation of cells based on these conditions is the job of the flattener, not `List` itself.